catalog/views.py

```diff
@@ -23,3 +24,9 @@ def index(request):
 
     # Render the HTML template index.html with the data in the context variable
     return render(request, 'index.html', context=context)
+
+class BookListView(generic.ListView):
```

Comment:
- Ngoài cách dùng generic ListView để lấy ra danh sách books thì còn cách nào khác không em?
- Trường hợp kết hợp một số điều kiện filter để lấy ra books thì làm cách nào? VD: các quyển sách mà tên có chứa ký tự 'B' chẳng hạn

Reply:
### 1. Generic ListView gọi là class-based views, ngoài cách này thì còn một cách là function-based views ạ. 
Tức là mình có thể query trực tiếp bên trong function giống hàm index ở trên và truyền danh sách đó qua context tới template tương ứng ạ
```
 def bookList(request):
    books = Book.objects.all()
    context = {
          'books': books,
    }
    return render(request, 'catalog/book_list.html', context=context)
```
Sau đó ở bên trong template mình có thể loop qua biến books để in ra từng quyển một ạ.

### 2. Trường hợp cần filter mình có 2 cách ạ:
- Thay đổi giá trị default của thuộc tính **queryset**
```
class BookListView(generic.ListView):
     model = Book
     queryset = Book.objects.filter(title__contains='B')
```
- Ghi đè phương thức get_queryset() của class
```
class BookListView(generic.ListView):
     model = Book

     def get_queryset(self):
        return Book.objects.filter(title__contains='B')
```
Cách 2 này flexible hơn cách 1 vì mình có thể tùy chỉnh nội dung query trong phạm vi function thay vì chỉ sửa 1 dòng code như cách 1 ạ.